netket/models/mlp.py

```diff
@@ -0,0 +1,94 @@
+from typing import Any, Tuple
+
+import numpy as np
+
+from flax import linen as nn
+from jax.nn.initializers import lecun_normal, zeros
+
+from netket.utils import deprecate_dtype
+from netket.utils.types import NNInitFunc
+from netket import nn as nknn
+
+default_kernel_init = lecun_normal()
+default_bias_init = zeros
+
+
+@deprecate_dtype
+class MLP(nn.Module):
+    r"""A Multi-Layer Perceptron"""
+    output_dim: int = 1
+    """The output dimension"""
+    hidden_dims: Tuple[int] = None
+    """The size of the hidden layers"""
+    alpha_hidden_dims: Tuple[int] = None
+    """The size of the hidden layers provided as number of times the input size"""
+    param_dtype: Any = np.float64
+    """The dtype of the weights."""
+    hidden_activation: Any = nknn.gelu
+    """The nonlinear activation function after each hidden layer."""
+    output_activation: Any = None
+    """The nonlinear activation at the output layer."""
```

Comment:
I prefer to separate output from hidden layers. It's more common to do it this way, since the output layer is typically different from the rest. For example, in MLPs for regression, the output activation is just linear, and for classification the output is typically as sigmoid or a softmax. But now you can specify also the activation function for all hidden layers separately.